Geometry/HcalCommonData/interface/HcalGeomParameters.h

```diff
@@ -0,0 +1,72 @@
+#ifndef HcalCommonData_HcalGeomParameters_h
+#define HcalCommonData_HcalGeomParameters_h
+
+/** \class HcalGeomParameters
+ *
+ * this class extracts some geometry constants from CompactView
+ * to be used by Reco Geometry/Topology
+ *  
+ *  $Date: 2015/06/25 00:06:50 $
+ * \author Sunanda Banerjee, Fermilab <sunanda.banerjee@cern.ch>
+ *
+ */
+
+#include<string>
+#include<vector>
+#include<iostream>
+
+#include "DetectorDescription/Core/interface/DDsvalues.h"
+#include "Geometry/HcalCommonData/interface/HcalCellType.h"
+#include "DataFormats/HcalDetId/interface/HcalSubdetector.h"
+
+class DDCompactView;    
+class DDFilteredView;
+
+class HcalGeomParameters {
+
+public:
+
+  HcalGeomParameters( const DDCompactView& cpv );
+  ~HcalGeomParameters();
+
+  std::vector<double>       getConstRHB()    const {return rHB;}
```

Comment:
@bsunanda - you can fill in HcalParameters record directly without caching the vectors here. Copying them seems unnecessary.